### PR TITLE
Optimize roads.mss

### LINKED ISSUE #366: Add maxzoom for layer state-names ?
Max zoom state-names is drawn depending of the size of the state.
State is for adminborder level 4 https://wiki.openstreetmap.org/wiki/Tag:boundary=administrative (region in France)
Though I didn't found example where they are drawn >z12, so I propose to set maxzoom at 13

Here sql result for z16 :
```
./test_sql.sh ../../project.mml --zoom 16 --db osm ....
+-----------+----------------+-----------------------------+-----------------+---------------------------+
| Time (ms) | #lines fetched |            Layer            | #fields fetched | Size (text format, in kB) |
+-----------+----------------+-----------------------------+-----------------+---------------------------+
|    3486   |     516278     |          roads_high         |        20       |           150688          |
|    1164   |     13432      |        amenities-poly       |        31       |            9878           |
|    1004   |     100487     |       amenities-points      |        31       |           10860           |
|    444    |      1931      |      highway_area_label     |        5        |            180            |
|    431    |      4874      |         landuse_gen1        |        4        |           19519           |
|    408    |     33865      |   roads-text-ref-low-zoom   |        6        |            8696           |
|    346    |     10092      |          area_label         |        5        |           10900           |
|    332    |      1380      |       landuse-overlay       |        5        |            7096           |
|    296    |      122       |          water_gen1         |        4        |            1232           |
|    292    |       15       |       military-overlay      |        5        |             19            |
|    291    |       36       |       protected-areas       |        6        |            765            |
|    287    |       1        |         state-names         |        5        |            511            |
|    284    |       0        |        country-names        |        0        |             0             |
|    260    |     12601      |            bridge           |        19       |            1994           |
|    221    |     19590      |            tunnel           |        19       |            4998           |
|    139    |      7615      |       placenames-small      |        7        |            661            |
|    118    |      1564      |        admin-mid-zoom       |        3        |           17190           |
|    106    |       1        |        capital-names        |        5        |             0             |
|    105    |      1433      | bicycle_routes_bicycle_gen3 |        5        |            2440           |
|    102    |      252       |      placenames-medium      |        6        |             18            |
|     86    |       1        |   bicycle_routes_mtb_gen3   |        5        |             2             |
|     85    |      1339      |    bicycle_routes_labels    |        7        |            2277           |
|     83    |       15       |         ferry-routes        |        2        |             9             |
|     74    |      1435      |         waterway_low        |        3        |            570            |
|     70    |      1266      |           aeroway           |        3        |            391            |
+-----------+----------------+-----------------------------+-----------------+---------------------------+
TOTAL: 25 layers in 10577ms with 729625 objets
```

### PR DESCRIPTION
**Pull request by Florimondable**
Result : building xml in 14s, 7,3Mo
Time rendering is also 10% lower, much less for z11-14
:)

Big improvements was :
remove 'road' type (make it as residential in sql)
Merge in sql can_bicycle=no|private. 
remove useless z11 inline zoom definition for minor roads
rewrite roads inline by type

**EDIT**: Closes #366.

### CODE REVIEW

File: project.mml [hunk 1]
```diff
@@ -748,13 +748,13 @@ Layer:
           CASE
             WHEN bicycle IS NOT NULL THEN bicycle
             WHEN tags->'motorroad' IN ('yes') THEN 'no'
-            WHEN highway NOT IN ('motorway', 'motorway_link') AND tags->'vehicle' IS NOT NULL THEN tags->'vehicle'
```
Comment (Phyks) [inline project.mml:751]: Why removing the `_link` highways?

File: roads.mss [hunk 2]
```diff
@@ -2277,25 +2277,20 @@
 // AEROWAYS
 // ==================================================================
 
-#aeroway[zoom>9] {
+#aeroway[zoom>11] {
```
Comment (Phyks) [inline roads.mss:2681]: Should be `zoom >= 11`.

File: roads.mss [hunk 3]
```diff
@@ -4,7 +4,7 @@
 
 // At lower zoomlevels, just show major automobile routes: motorways
 // and trunks.
-
+/**/
```
Comment (Phyks) [inline roads.mss]: Temp comment left?

File: roads.mss [hunk 4]
```diff
@@ -1279,93 +1279,15 @@
   [type='pedestrian']   { marker-transform: translate(-0.5 * @rdz18_pedestrian - 0.5 * @rdz18_cycle); }
 }
 
-
-// ---- Bridge fill for dashed lines --------------------------------
-#tunnel::line[zoom>=14],
-#bridge::line[zoom>=14] {
```
Comment (Phyks) [inline roads.mss:1300]: I'm fine with this for tunnel. But isn't it an issue for bridge?

File: roads.mss [hunk 5]
```diff
@@ -1303,150 +1259,1060 @@
     }
   }
 }
-
+/**/
```
Comment (Phyks) [inline roads.mss:1262]: Temp comment left?

File: project.mml [hunk 6]
```diff
@@ -522,9 +522,12 @@ Layer:
             ELSE NULL
           END AS cycleway_right_oneway,
           CASE
+            WHEN bicycle IN ('no', 'private') THEN 'no'
```
Comment (Phyks) [inline project.mml:525]: Minor comment, but might be better to merge into a `no_or_private` value or something like this? (for readability)

File: roads.mss [hunk 7]
```diff
@@ -2395,4 +2387,4 @@
   [zoom >= 18] {
     line-width: 14;
   }
-}
+}/**/
```
Comment (Phyks) [inline roads.mss]: Extra comments?

File: project.mml (hunk 1 above)
Comment (Florimondable) [reply project.mml:751]: useless below z11

File: roads.mss (hunk 7 above)
Comment (Florimondable) [reply roads.mss]: a comment trick to comment everything by just adding `/*` at the top

File: roads.mss (hunk 2 above)
Comment (Florimondable) [reply roads.mss:2681]: yes

File: roads.mss (hunk 4 above)
Comment (Florimondable) [reply roads.mss:1300]: No, roads have plain inline color

File: project.mml (hunk 6 above)
Comment (Florimondable) [reply project.mml:525]: can_bicycle is already a computed variable

**Pull request merged**